src/NationalInstruments.Analyzers/Style/ChainOfMethodsWithLambdasAnalyzer.cs

```diff
@@ -0,0 +1,110 @@
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using NationalInstruments.Analyzers.Properties;
+using NationalInstruments.Analyzers.Utilities;
+using NationalInstruments.Analyzers.Utilities.Extensions;
+
+namespace NationalInstruments.Analyzers.Style
+{
+    /// <summary>
+    /// Enforces the convention -
+    /// https://github.com/ni/csharp/blob/main/docs/Coding-Conventions.md#f34-%EF%B8%8F-consider-splitting-long-chains-of-dotted-methodproperty-invocations
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class ChainOfMethodsWithLambdasAnalyzer : NIDiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "NI1017";
+
+        public static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+            DiagnosticId,
+            new LocalizableResourceString(nameof(Resources.NI1017_Title), Resources.ResourceManager, typeof(Resources)),
+            new LocalizableResourceString(nameof(Resources.NI1017_Message), Resources.ResourceManager, typeof(Resources)),
+            Resources.CategoryNI,
+            DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecutionIf(IsRunningInProduction);
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterSyntaxNodeAction(
+                AnalyzeSyntaxNode,
+                SyntaxKind.ExpressionStatement,
+                SyntaxKind.EqualsValueClause,
+                SyntaxKind.Argument,
+                SyntaxKind.ArrowExpressionClause);
+        }
+
+        private static void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context)
+        {
+            var parentSyntaxNode = context.Node;
+
+            // Find the invocation expression i.e
+            // a method/delegate call or a property access
+            // or a chain of them
+            var invocationExpressionSyntax = parentSyntaxNode
+                .DescendantNodes()
+                .OfType<InvocationExpressionSyntax>()
+                .FirstOrDefault();
+
+            if (invocationExpressionSyntax is null)
+            {
+                // This does not contain any invocations, bail out
+                return;
+            }
+
+            bool IsLambdaExpression(SyntaxNode syntaxNode) => syntaxNode.IsKind(SyntaxKind.SimpleLambdaExpression);
```

Comment:
Local functions should always go at the bottom

Reply:
Also, why not search for SimpleLambdaExpressions right off the bat?

Reply:
> Local functions should always go at the bottom

Done. We need an analyzer for this! 😆 

> Also, why not search for SimpleLambdaExpressions right off the bat?

Straight up searching for a `SimpleLambdaExpression` would return me all the lambdas with the expression, even the **nested ones**. For e.g. this statement...

```cs
xyz.GroupBy(software => software.AliasName).Select(g => g.First(x => x != null))
```

...would return three lambdas whereas I really want two.